src/lib/comms/sol-oic-security.c

```diff
@@ -266,11 +269,26 @@ creds_init(const void *data)
     creds->security = data;
     sol_vector_init(&creds->items, sizeof(struct cred_item));
 
-    file_data = sol_util_load_file_string(file_name, &length);
-    if (!file_data)
+    cert = sol_cert_load_from_file(file_name);
```

Comment:
this is not right as it implies a file. This logic should move away from sol-oic-security into sol-cert. For instance "sol_cert_new_for_XXX()", like "XXX" being  "system", "app" or "user", depends on the scope of it, that would result saving in "/etc/ssl", "~/.config/APPNAME" or "~/.ssl" (or something along these lines, need to check the XDG convention for applications and the openssl for system and user.

Then, internally, these will create a path and load or save it. In Small OS they would be the same function, since there are no separation.